### PR TITLE
docs: Update Pull Request generator documentation to include application lifecycle

### PR DESCRIPTION
**Pull request by Geethree**
The lifecycle of the generated applications was not explained as discussed [here](https://cloud-native.slack.com/archives/C01U45M2SVB/p1668194689295539?thread_ts=1668194193.486229&cid=C01U45M2SVB) in slack.

Happy to modify, mainly throwing this up before I forget about it.

Signed-off-by: Gerald Spencer <Geethree@users.noreply.github.com>

### CODE REVIEW

File: docs/operator-manual/applicationset/Generators-Pull-Request.md [hunk 1]
```diff
@@ -318,3 +318,7 @@ The Pull Request Generator will requeue when the next action occurs.
 - `merge`
 
 For more information about each event, please refer to the [official documentation](https://docs.gitlab.com/ee/user/project/integrations/webhook_events.html#merge-request-events).
+
+## Lifecycle
+
+An appliction will be generated when a Pull Request is discovered when the configured crietia is met - i.e. for Github when a Pull request matches the specified `labels` and/or `pullRequestState`. Applications will be removed when a Pull Request no longer meets the specified criteria.
```
Comment (alexef) [inline docs/operator-manual/applicationset/Generators-Pull-Request.md:324]:
```suggestion
An Application will be generated when a Pull Request is discovered when the configured criteria is met - i.e. for GitHub when a Pull Request matches the specified `labels` and/or `pullRequestState`. Application will be removed when a Pull Request no longer meets the specified criteria.
```

**Review by alexef — CHANGES_REQUESTED** [review]
typos

**Comment by Geethree** [comment]
@alexef Any other additions you'd like to see made?

**Review by crenshaw-dev — APPROVED** [review]

**Comment by crenshaw-dev** [comment]
Cherry-picked onto release-2.5 for 2.5.4. Thanks for your patience, @Geethree!

**Pull request merged**